src/rust/ide/src/controller/graph/executed.rs

```diff
@@ -141,7 +147,20 @@ impl Handle {
         let value_stream = registry.subscribe().map(Notification::ComputedValueInfo).boxed_local();
         let graph_stream = self.graph().subscribe().map(Notification::Graph).boxed_local();
         let self_stream  = self.notifier.subscribe().boxed_local();
-        futures::stream::select_all(vec![value_stream,graph_stream,self_stream])
+
+        // Note: [Argument Names-related invalidations]
+        let db_stream = self.project.suggestion_db().subscribe().map(|notification| {
+            match notification {
+                model::suggestion_database::Notification::Updated =>
+                    Notification::Graph(controller::graph::Notification::Invalidate),
+            }
+        }).boxed_local();
+        let update_stream = registry.subscribe().map(|_| {
+            Notification::Graph(controller::graph::Notification::Invalidate)
+        }).boxed_local();
+
+        let streams = vec![value_stream,graph_stream,self_stream,db_stream,update_stream];
+        futures::stream::select_all(streams)
```

Comment:
```suggestion
        let streams = [value_stream,graph_stream,self_stream,db_stream,update_stream];
        futures::stream::select_all(streams)
```
In theory there are no reason why passing statically-sized array should not work here, however for Rust it may be not so obvious. But you may check if you want.